src/Analysis/Ast/Impl/Analyzer/ModuleWalker.cs

```diff
@@ -15,22 +15,36 @@
 
 using System.Diagnostics;
 using System.Linq;
+using Microsoft.Python.Analysis.Analyzer.Evaluation;
 using Microsoft.Python.Analysis.Documents;
 using Microsoft.Python.Analysis.Modules;
 using Microsoft.Python.Analysis.Types;
 using Microsoft.Python.Analysis.Values;
 using Microsoft.Python.Core;
+using Microsoft.Python.Core.Collections;
 using Microsoft.Python.Core.Diagnostics;
 using Microsoft.Python.Parsing.Ast;
 
 namespace Microsoft.Python.Analysis.Analyzer {
     [DebuggerDisplay("{Module.Name} : {Module.ModuleType}")]
     internal class ModuleWalker : AnalysisWalker {
+        private const string AllVariableName = "__all__";
         private readonly IDocumentAnalysis _stubAnalysis;
 
+        // A hack to use __all__ export in the most simple case.
+        private int _allReferencesCount;
+
         public ModuleWalker(IServiceContainer services, IPythonModule module, PythonAst ast)
-            : base(services, module, ast) {
+            : base(new ExpressionEval(services, module, ast)) {
             _stubAnalysis = Module.Stub is IDocument doc ? doc.GetAnyAnalysis() : null;
+            ExportedMemberNames = ImmutableArray<string>.Empty;
+        }
+
+        public override bool Walk(NameExpression node) {
```

Comment:
This would mark uses; it's probably better to check for it in an assignment or augmented assignment node's left.

Reply:
It can be also updated using `__all__.append(...)`, so I have to count simple uses as well.